pyvista/plotting/plotter.py

```diff
@@ -3331,7 +3331,15 @@ def add_mesh(
                 raise TypeError(
                     f'Object type ({type(mesh)}) not supported for plotting in PyVista.'
                 )
-        if isinstance(mesh, pyvista.PointSet):
+        if isinstance(mesh, pyvista.PolyData):
```

Comment:
@MatthewFlamm is right. @Arnav-2004 Do you get what they said? You are welcome anything to ask.

Reply:
Although I think my above comment includes all the key pointers needed to fix this without impacting long existing features, it is dense for a newer contributor.  Please ask any clarifying questions needed to get this to work.  If you don't feel comfortable working on the documentation in multiple spots, we can help add that.

Reply:
I am impressed that he was able to identify the cause here in his first contribution.

Reply:
> This will transform _all_ `PolyData`. We shouldn't do this as it will be unexpected for every other use case besides this one.

I understand, but as stated in https://github.com/pyvista/pyvista/issues/4956#issuecomment-1742763166 if I check for `isinstance(mesh, _vtk.vtkCylinderSource)` the condition evaluates to True but it doesn't seem to be rotating it, maybe I need to have a better understanding of how the `vtkCylinderSource` class works

Reply:
nits. `isinstance(mesh, pv.CylinderSource)` works fine. We try to hide vtk objects as much as possible in code.

Reply:
As for the rotation, I think it would work fine with what you removed in the `output` method. I am making a very difficult request and welcome any confirmation questions.

Reply:
My earlier comment was that we should not do any special casing inside plotting and make the pyvista output method object exactly match the algorithm output.  Otherwise, we will end up with an increasingly complex web of rules for plotting.  What if someone consumes a CylinderSource in the middle of a whole pipeline as an example of something that might happen in the future?

Reply:
Sorry for my misunderstanding. I'm lazy and all I was thinking about was simplifying the process on the plotting side. We certainly need to think this through.

Reply:
@MatthewFlamm 
I think it is a good idea to change the function `algorithm_to_mesh_handler` so that if the object has an output method, it uses that instead of the `GetOutputPort `method. This would require minimal modification on the part of `Plotter`, and we can make it a rule that future algorithm classes should implement the `output ` method uniformly. Could you give your thoughts?

Reply:
> We need to move these rotate and translate lines back into  `Cylinder` from `geometric_object.py`.  Unfortunately, pyvista did not use the default behavior from vtk for `Cylinder` and now it is inconsistent.  In my opinion, it is more preferable that `CylinderSource` is internally consistent.  We should document this difference in the documentation of both `Cylinder` and `CylinderSource`.

Sorry to confuse the discussion, but I have moved to the opinion that it is better to follow this, given the confusion when using `GetOutputPort` directly.

Reply:
@Arnav-2004
1. Please move the rotating code line to `Cylinder` function.
2. Let's write document that says `Cylinder` and `CylinderSource` position is different with us.

Reply:
As per my understanding, right now I just need to move the rotating and translate logic into the `Cylinder` function, and document the position difference when using `Cylinder` and `CylinderSource`. Overall, do I need to produce the same orientation or not?

Reply:
@Arnav-2004 You don't need to between `Cylinder` and  `CylinderSource.` Sorry for confusing.

Reply:
@tkoyama010 I have encountered a new problem. After moving the rotate and translate logic inside the `Cylinder` the orientation of `CylinderSource` is getting changed.

| cyl_source.output.plot() | pl.add_mesh(cyl_source) |
| -------------------------  | --------------------------  |
| ![image](https://github.com/pyvista/pyvista/assets/70969910/5c745ddd-0fac-41cc-9f3c-81ecdb737a6c) | ![image](https://github.com/pyvista/pyvista/assets/70969910/b4930022-fe03-4441-a2a3-caf3a35f9685) |

Reply:
@Arnav-2004
That is weird. Could you commit and push your current status? Then we can review it.

Reply:
@tkoyama010 this is after the fact I remove 
```py
if isinstance(mesh, pyvista.PolyData):
        mesh_points = mesh.points.copy()
        angle = np.radians(90.0)
        rotation_matrix = np.array([
            [np.cos(angle), -np.sin(angle), 0],
            [np.sin(angle), np.cos(angle), 0],
            [0, 0, 1]
        ])
        rotated_points = np.dot(mesh_points, rotation_matrix)
        mesh.points = rotated_points
```

Reply:
@Arnav-2004 You are right. Did you then add the lines to the Cylinder function?

Reply:
@tkoyama010 the problem is, if I add this logic again inside the Cylinder function the orientation of the following will change:

```py
>>> import pyvista
>>> import numpy as np
>>> cylinder = pyvista.Cylinder(
...     center=[1, 2, 3], direction=[1, 1, 1], radius=1, height=2
... )
>>> cylinder.plot(show_edges=True, line_width=5, cpos='xy')
```

Reply:
@Arnav-2004
Sorry I am not aware of the situation. Is the result after the change the same as the result of `pyvista==0.42.2`? The purpose is to match the results with the previous version, so if they are the same, it's fine.

I briefly checked your changes and they look correct.

Reply:
@tkoyama010 After the current latest change, the orientation of: `cyl_source.output.plot()` changes to

![image](https://github.com/pyvista/pyvista/assets/70969910/7b8f971c-dbc4-45f0-afc5-0bb8428c25c0)

Reply:
@Arnav-2004 
Sorry, I was not explaining it well. Our goal is to have the same `PolyData` derived from `CylinderSource`.

### Plotting of `CylinderSource` object
These three plots same.
1. `pv.CylinderSource().output.plot()`
1. `pl.add_mesh(pv.CylinderSource().output)`
1. `pl.add_mesh(pv.CylinderSource())`

### Plotting of `Cylinder` function
These two plots same.
1. `pl.add_mesh(pv.Cylinder())`
1. `pv.Cylinder().plot()`

The specification is that the function Cylinder rotates the CylinderSource's PolyData in its own way. So the plot of CylinderSource class and Cylinder function is different. So we will explain about it in each documentation which we will add.

I hope it makes sense to you.

Reply:
Sorry if this is taking longer than expected. But I get what you mean now, will get started on documenting that.

Reply:
@Arnav-2004 
Thank you! That is a challenging area and we will provide support as needed.